### DISCUSSION TITLE
Field Array simple required/min array length validating

### DISCUSSION THREAD

**Discussion by douglasrcjames** [react-hook-form/react-hook-form#8869] [Q&A] [answered]
Hi! First off, I love this library, one of my top 5 favorite right now! 

This may be an "Idea" for a change, but I am trying to validate a field array before submission is required, just as simple it has at least 1 field in the array. I attempted to use Yup validation (see https://github.com/react-hook-form/react-hook-form/discussions/4475) just for one validation. I couldn't get the resolver to work with my code structure, maybe because I am not using Yup for my other validations. I'd like to not use Yup as a 3rd party library for such a small usage, nor rebuild my other validations for yup schema usage. 

Before I was trying Yup, I was following the [documentation](https://react-hook-form.com/api/usefieldarray), and used the `rules` prop like below: 
```
  const fieldArray = useFieldArray({
      control: editItemForm.control,
      name: "test1",
      rules: {
          required: "You need to create a field.",
          minLength: 1,
          maxLength: 10
      },
  });

// also how I am validating within the input element
<TextInput
    type="text"
    error={error}
    placeholder={`Enter a short text value.`}
    {
        ...editItemForm.register(column.key, column.validators)
    }
/>
```
What is this `rules` validation there for? I can't seem to get it to throw any errors from `editItemForm.formState.errors`.

Since it seems this feature doesn't exist, it would be amazing if the library's `useFieldArray` function could have a prop (like I thought the above `rules` prop might work) that you can pass a rules to limit the maximum and minimum length of the field array.  I feel like this is a very common use case for people using FieldArray, for it to be "optional" or "required" to have at least 1 entry, and using a full different 3rd party library for such a small feature, isn't ideal. Please let me know if I misunderstand how to achieve this goal!

**Comment by Moshyfawn** [comment, marked as answer]
You should check what RHF version you're using. The `rules` API for `useFieldArray` hook just arrived in [v7.34.0](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.34.0)

Additionally, you won't find your **root** field array message on the main `formState.errors` object, but on `errors.fields?.root?.message`.

You can read [the `useFieldArray` docs section](https://react-hook-form.com/api/usefieldarray) to learn more.

Check out [this working csb](https://codesandbox.io/s/nice-tesla-6kvy1t?file=%2Fsrc%2FApp.tsx%3A1842-1861) as well.

**Reply by Moshyfawn** [reply]
Also, thx for the kind words about the lib ;)

**Reply by douglasrcjames** [reply]
This worked, thank you!!
